pkg/login/login.go

```diff
@@ -0,0 +1,80 @@
+package login
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/dcos/dcos-cli/pkg/httpclient"
+	"github.com/sirupsen/logrus"
+)
+
+// Client is able to detect available login providers and login to DC/OS.
+type Client struct {
+	http   *httpclient.Client
+	logger *logrus.Logger
+}
+
+// NewClient creates a new login client.
+func NewClient(baseClient *httpclient.Client, logger *logrus.Logger) *Client {
+	return &Client{
+		http:   baseClient,
+		logger: logger,
+	}
+}
+
+// Providers returns the supported login providers for a given DC/OS cluster.
+func (c *Client) Providers() (map[string]*Provider, error) {
+	resp, err := c.http.Get("/acs/api/v1/auth/providers")
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	providers := make(map[string]*Provider)
+	if resp.StatusCode == 200 {
+		err := json.NewDecoder(resp.Body).Decode(&providers)
+		if err == nil {
+			return providers, nil
+		}
+	}
+
+	c.logger.Info("Falling back to the WWW-Authenticate challenge.")
+	authHeader, err := c.challengeAuth()
+	if err != nil {
+		return nil, err
+	}
+
+	switch authHeader {
+	case "oauthjwt":
+		providers["dcos-oidc"] = defaultOIDCImplicitFlowProvider()
```

Comment:
@jgehrcke : Ideally this will match the future content of the login providers endpoint in DC/OS OSS, especially for the provider ID. It currently uses : 
``` json 
{
        "dcos-oidc": {
                "authentication-type": "oidc-implicit-flow",
                "client-method": "browser-prompt-authtoken",
                "config": {
                        "start_flow_url": "/login?redirect_uri=urn:ietf:wg:oauth:2.0:oob"
                },
                "description": "Google, GitHub, or Microsoft"
        }
}
```

Let me know if I should adjust here.

Reply:
Follow-up from JP : 
> `dcos-oidc-auth0`
that identity provider is specifically Auth0
serves as a "proxy" to other identity providers